### PR TITLE
[LW] Fix cache size metrics

### PR DESCRIPTION
**Pull request by Jolyon-S**
**Goals (and why)**:
Fix cache metrics - they aren't reset when the cache is reset.

**Implementation Description (bullets)**:
Reset metrics on reset.

**Concerns (what feedback would you like?)**:
Is `cleanUp()` too heavy-handed?

**Where should we start reviewing?**:
Diff

**Priority (whenever / two weeks / yesterday)**:
Whenever

### CODE REVIEW

File: atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/keyvalue/api/cache/CacheMetrics.java [hunk 1]
```diff
@@ -87,6 +87,10 @@ public void decreaseCacheSize(long removed) {
         cacheSize.dec(removed);
     }
 
+    public void resetCacheSize() {
+        cacheSize.dec(cacheSize.getCount());
+    }
```
Comment (Jolyon-S) [inline atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/keyvalue/api/cache/CacheMetrics.java:92]: why there isn't a reset method is beyond me

File: timelock-server/src/integTest/java/com/palantir/atlasdb/timelock/LockWatchValueIntegrationTest.java [hunk 2]
```diff
@@ -406,7 +406,7 @@ public void nearbyCommitsDoNotAffectResultsPresentInCache() {
         AtomicReference<LockWatchCache> lwCache = new AtomicReference<>(null);
         PreCommitCondition condition = timestamp -> {
             if (timestamp != startTs.get()) {
-                simulateOverlappingWriteTransaction(lwCache, timestamp);
+                simulateOverlappingWriteTransaction(lwCache, startTs.get(), timestamp);
```
Comment (Jolyon-S) [inline timelock-server/src/integTest/java/com/palantir/atlasdb/timelock/LockWatchValueIntegrationTest.java:409]: Changes in this class are a drive-by to reduce flakes (there was a weird race condition here, because I think other transactions were being started in the background).

File: atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/keyvalue/api/cache/CacheMetrics.java (hunk 1 above)
Comment (gmaretic) [reply atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/keyvalue/api/cache/CacheMetrics.java:92]: Probably they expect you to throw it away and create a new counter?

File: atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/keyvalue/api/cache/ValueStoreImpl.java [hunk 3]
```diff
@@ -78,8 +78,12 @@
     @Override
     public void reset() {
         values.resetToInitialValue();
-        loadedValues.invalidateAll();
         watchedTables.resetToInitialValue();
+        loadedValues.invalidateAll();
+
+        // Forcing the cache to run cleanup here guarantees that the metrics are not affected after they have been reset
+        loadedValues.cleanUp();
+        metrics.resetCacheSize();
```
Comment (gmaretic) [inline atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/keyvalue/api/cache/ValueStoreImpl.java:86]: Kind of jank that invalidation does not trigger removal listeners

**Review by gmaretic — APPROVED** [review]
LGTM
